test/starknet/MerkleWhitelist.test.ts

```diff
@@ -0,0 +1,125 @@
+      .sort(function (a, b) {
+        if (a > b) return 1;
+        if (a < b) return -1;
+        return 0;
+      })
+      .map((x, i) => [x[0], x[1], x[2], i]);
+    leaves = merkleData.map((x) => x[0].toString());
+    tree = new MerkleTree(leaves);
+  });
+
+  it('returns the voting power for everyone in the list', async () => {
+    const leaf1 = merkleData.find((leaf) => leaf[1] == address1)!;
+    const { voting_power: vp1 } = await whitelist.call('getVotingPower', {
+      timestamp: BigInt(0),
+      voter_address: { value: address1 },
+      params: [tree.root],
+      user_params: [address1, power1.low, power1.high, ...tree.getProof(leaves, Number(leaf1[3]))],
+    });
+    const leaf2 = merkleData.find((leaf) => leaf[1] == address2)!;
+    const { voting_power: vp2 } = await whitelist.call('getVotingPower', {
+      timestamp: BigInt(0),
+      voter_address: { value: address2 },
+      params: [tree.root],
+      user_params: [address2, power2.low, power2.high, ...tree.getProof(leaves, Number(leaf2[3]))],
+    });
+    const leaf3 = merkleData.find((leaf) => leaf[1] == address3)!;
+    const { voting_power: vp3 } = await whitelist.call('getVotingPower', {
+      timestamp: BigInt(0),
+      voter_address: { value: address3 },
+      params: [tree.root],
+      user_params: [address3, power3.low, power3.high, ...tree.getProof(leaves, Number(leaf3[3]))],
+    });
+    const leaf4 = merkleData.find((leaf) => leaf[1] == address4)!;
+    const { voting_power: vp4 } = await whitelist.call('getVotingPower', {
+      timestamp: BigInt(0),
+      voter_address: { value: address4 },
+      params: [tree.root],
+      user_params: [address4, power4.low, power4.high, ...tree.getProof(leaves, Number(leaf4[3]))],
+    });
+
+    expect(
+      new utils.splitUint256.SplitUint256(`0x${vp1.low.toString(16)}`, `0x${vp1.high.toString(16)}`)
+    ).to.deep.equal(power1);
+    expect(
+      new utils.splitUint256.SplitUint256(`0x${vp2.low.toString(16)}`, `0x${vp2.high.toString(16)}`)
+    ).to.deep.equal(power2);
+    expect(
+      new utils.splitUint256.SplitUint256(`0x${vp3.low.toString(16)}`, `0x${vp3.high.toString(16)}`)
+    ).to.deep.equal(power3);
+    expect(
+      new utils.splitUint256.SplitUint256(`0x${vp4.low.toString(16)}`, `0x${vp4.high.toString(16)}`)
+    ).to.deep.equal(power4);
+  }).timeout(1000000);
+
+  it('Should fail if an invalid proof is supplied', async () => {
+    const leaf1 = merkleData.find((leaf) => leaf[1] == address1)!;
+    try {
+      const { voting_power: vp } = await whitelist.call('getVotingPower', {
+        timestamp: BigInt(0),
+        voter_address: { value: address2 },
+        params: [tree.root],
+        user_params: [
+          address1,
+          power1.low,
+          power1.high,
+          ...tree.getProof(leaves, Number(leaf1[3])),
+        ],
+      });
+    } catch (err: any) {
+      expect(err.message).to.contain('MerkleWhitelist: Invalid proof supplied');
+    }
+  }).timeout(1000000);
```

Comment:
would add test to prove a leaf that is not in the tree

Reply:
proof of non-membership isnt required here though

Reply:
Sure! But making sure it fails would be nice :)